smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/TypeScriptJmesPathVisitor.java

```diff
@@ -0,0 +1,334 @@
+        String rightContext = executionContext;
+
+        executionContext = String.format("(%s %s %s)", leftContext, comparator, rightContext);
+        return null;
+    }
+
+    @Override
+    public Void visitCurrentNode(CurrentExpression expression) {
+        // Fall through as visitCurrentNode is saying that there is a noop here. Execution context does not change.
+        return null;
+    }
+
+    @Override
+    public Void visitExpressionType(ExpressionTypeExpression expression) {
+        throw new CodegenException("TypeScriptJmesPath visitor not implemented ExpressionTypeExpression");
+    }
+
+    @Override
+    public Void visitFlatten(FlattenExpression expression) {
+        expression.getExpression().accept(this);
+        String flatScope = makeNewScope("flat_");
+        writer.write("let $L: any[] = [].concat(...$L);", flatScope, executionContext);
+        executionContext = flatScope;
+        return null;
+    }
+
+    @Override
+    public Void visitFunction(FunctionExpression expression) {
+        ArrayList<String> executionContexts = new ArrayList<>();
+
+        expression.arguments.forEach((JmespathExpression argExpression) -> {
+            argExpression.accept(this);
+            switch (expression.getName()) {
+                case "length":
+                    executionContext = executionContext + ".length";
+                    break;
+                case "contains":
+                    executionContexts.add(executionContext);
+                    break;
+                default:
+                    throw new CodegenException("TypeScriptJmesPath visitor has not implemented function: "
+                            + expression.getName());
+            }
+        });
+
+        if (expression.getName().equals("contains")) {
+            executionContext = String.join(".includes(", executionContexts) + ")";
+        }
+        return null;
+    }
+
+    @Override
+    public Void visitField(FieldExpression expression) {
+        executionContext += ".";
+        executionContext += expression.getName();
+        return null;
+    }
+
+    @Override
+    public Void visitIndex(IndexExpression expression) {
+        if (expression.getIndex() >= 0) {
+            executionContext += ("[" + expression.getIndex() + "]");
+        } else {
+            executionContext += "[" + executionContext + ".length";
+            executionContext += " - " + Math.abs(expression.getIndex()) + "]";
+        }
+        return null;
+    }
+
+    @Override
+    public Void visitLiteral(LiteralExpression expression) {
+        switch (expression.getType()) {
+            case STRING:
+                executionContext = "\"" + expression.getValue().toString() + "\"";
+                break;
+            default:
+                // All other options are already valid js literials.
+                // (BOOLEAN, ANY, ARRAY, NULL, NUMBER, OBJECT, EXPRESSION)
+                executionContext = expression.getValue().toString();
```

Comment:
I guess you're expecting a JSON encoding here, right? You need to do this instead:

```
executionContext = Node.printJson(expression.getValue());
```

Reply:
I need to cast to node, but I think this should work

Reply:
Actually just removed resolution of Arrays and objects for now. They dont show up in any waiters, defering this work post GA